modules/core/02-client/keeper/client.go

```diff
@@ -120,6 +128,27 @@ func (k *Keeper) UpdateClient(ctx sdk.Context, clientID string, clientMsg export
 	return nil
 }
 
+func (k *Keeper) CheckTxUpdateClient(ctx sdk.Context, clientID string, clientMsg exported.ClientMessage) error {
+	if status := k.GetClientStatus(ctx, clientID); status != exported.Active {
+		return errorsmod.Wrapf(types.ErrClientNotActive, "cannot update client (%s) with status %s", clientID, status)
+	}
+
+	_, clientModule, err := k.getClientTypeAndModule(clientID)
+	if err != nil {
+		return err
+	}
+
+	if !ctx.IsReCheckTx() {
+		if err := clientModule.VerifyClientMessage(ctx, clientID, clientMsg); err != nil {
+			return err
+		}
+	}
+
+	_ = clientModule.UpdateState(ctx, clientID, clientMsg)
```

Comment:
needs: https://github.com/cosmos/ibc-go/pull/6276 to avoid an edge case where misbehaviour would panic on antes and never get in